lib/redux/middleware/sentry_api_middleware.dart

```diff
@@ -15,49 +16,43 @@ class SentryApiMiddleware extends MiddlewareClass<AppState> {
           final Set<String> projectIdsWithSessions = {};
+          final Map<String, List<Project>> projectsByOrganizationId = {};
+
+          var fullProgress = organizations.length * 2;
+          var currentProgress = 0;
 
           for (final organization in organizations) {
             final individualOrganization = await api.organization(organization.slug);
             individualOrganizations.add(individualOrganization ?? organization);
-            final currentCursor = store.state.globalState.projectCursorsByOrganizationSlug != null && !action.reload
-                ? store.state.globalState.projectCursorsByOrganizationSlug[organization.slug]
-                : null;
 
-            final nextCursor = action.pagination
-                ? currentCursor == null
-                ? Cursor(10, 0, 0)
-                : Cursor(10, currentCursor.offset + 1, 0)
-                : null;
-
-            final projects = await api.projects(organization.slug, nextCursor);
+            store.dispatch(LoadingAction(true, '${organization.name} -> Fetching projects...', ++currentProgress / fullProgress));
+            final projects = await api.projects(organization.slug);
             if (projects.isNotEmpty) {
               projectsByOrganizationId[organization.slug] = projects;
             }
 
-            final projectsWithSessionsForOrganization = await api.projectIdsWithSessions(organization.slug);
-            projectIdsWithSessions.addAll(projectsWithSessionsForOrganization);
-
-            if (action.pagination) {
-              // Keep cursor if there are less than value projects
-              if (projects.length < nextCursor.value) {
-                projectCursorsByOrganizationSlug[organization.slug] = currentCursor;
-              } else {
-                projectCursorsByOrganizationSlug[organization.slug] = nextCursor;
-              }
+            store.dispatch(LoadingAction(true, '${organization.name} -> Checking for sessions...', ++currentProgress / fullProgress));
+            try {
+              final projectsWithSessionsForOrganization = await api.projectIdsWithSessions(organization.slug);
+              projectIdsWithSessions.addAll(projectsWithSessionsForOrganization);
+            } catch (e) {
+              Sentry.addBreadcrumb(Breadcrumb(message: 'Org has no projects -> $e', level: SentryLevel.error));
             }
           }
-          store.dispatch(FetchOrganizationsAndProjectsSuccessAction(individualOrganizations, projectsByOrganizationId, projectCursorsByOrganizationSlug, projectIdsWithSessions, action.reload));
+          store.dispatch(LoadingAction(true, 'Almost there'));
+          store.dispatch(FetchOrganizationsAndProjectsSuccessAction(individualOrganizations, projectsByOrganizationId, projectIdsWithSessions));
         } catch (e, s) {
           store.dispatch(FetchOrganizationsAndProjectsFailureAction(e, s));
         }
+        store.dispatch(LoadingAction(false));
```

Comment:
Since loading action is specific for this screen, we could remove the boolean parameter and just provide the progress and text, since it is clear that we are loading between `FetchOrganizationsAndProjectsAction` and `FetchOrganizationsAndProjectsSuccessAction/FetchOrganizationsAndProjectsFailureAction `. This would remove some redundancy.

Reply:
I would just keep the `boolean` so we don't need to deal with the "nullability" of text and progress. Only when the boolean is true we show the loading screen.

Reply:
Could have written this better. 🤦  I meant that `FetchOrganizationsAndProjectsAction` and `FetchOrganizationsAndProjectsSuccessAction/FetchOrganizationsAndProjectsFailureAction` would set the boolean, as the represent the flow/lifecycle of the call. Either way works in the end, so we can also leave it in the loading action.